dynamiqs/mesolve/rouchon.py

```diff
@@ -20,26 +19,18 @@ def __init__(self, *args, **kwargs):
         self.I = torch.eye(self.n, device=self.H.device, dtype=self.H.dtype)  # (n, n)
         self.dt = self.options.dt
 
-        self.M1s = sqrt(self.dt) * self.jump_ops  # (1, len(jump_ops), n, n)
-        self.M1s_adj = self.M1s.adjoint()  # (1, len(jump_ops), n, n)
-
 
 class MERouchon1(MERouchon):
-    @depends_on_H
-    def Hdag_nh(self, t: float) -> Tensor:
-        # -> (b_H, 1, n, n)
-        return self.H_nh(t).adjoint()
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-    @depends_on_H
-    def M0(self, t: float, dt: float) -> Tensor:
-        # build time-dependent Kraus operators
-        # -> (b_H, 1, n, n)
-        return self.I - 1j * dt * self.H_nh(t)
+        # define cached operators
+        self.M0 = cache(lambda H_nh: self.I - 1j * self.dt * H_nh)  # (b_H, 1, n, n)
+        self.M0dag = cache(lambda M0: M0.adjoint())  # (b_H, 1, n, n)
+        self.M0inv = cache(lambda H_nh: self.I + 1j * self.dt * H_nh)  # (b_H, 1, n, n)
```

Comment:
`M0inv` is not good notation here (you're not defining the inverse). Maybe `M0rev` instead? as in "reverse"